exporter/exporter.go

```diff
@@ -72,9 +72,17 @@ type Factory interface {
 	// LogsStability gets the stability level of the Logs exporter.
 	LogsStability() component.StabilityLevel
 
+	// Metadata returns the metadata describing the receiver.
+	Metadata() Metadata
```

Comment:
I wonder if this should have a different name. The metadata name suggests to me that the properties listed are properties of the component itself, but this is not the case: the service is the one that uses this information to decide how to build the graph. I see this more as a 'hint' than 'metadata'?

Reply:
I was thinking that since it's a method on the factory, it describes the type of component that the factory produces. Although the immediate usage is in the service, I assume we might use it elsewhere, e.g. documentation or cli output.

That said, I don't feel strongly about the name. Hints makes sense to me too, just seems a little more obscure.

Reply:
Re-reading the PR what I said does not make much sense: the service only handles logging, which it should, and building the component as a singleton is left to the component internals itself.

I don't feel strongly about the name, if you feel like `Metadata` is okay I am okay with that.